nil/internal/execution/block_generator.go

```diff
@@ -174,21 +174,21 @@ func (g *BlockGenerator) GenerateZeroState(zeroStateYaml string, config *ZeroSta
 	return res.Block, nil
 }
 
-func (g *BlockGenerator) prepareExecutionState(proposal *Proposal, logger zerolog.Logger) error {
+func (g *BlockGenerator) prepareExecutionState(proposal *Proposal) error {
 	if g.executionState.PrevBlock != proposal.PrevBlockHash {
 		esJson, err := g.executionState.MarshalJSON()
 		if err != nil {
-			logger.Err(err).Msg("Failed to marshal execution state")
+			g.logger.Err(err).Msg("Failed to marshal execution state")
```

Comment:
Such changes are really good and I propose to move them into separate patch to merge out of scope of this review.

Reply:
well, it's just a couple of lines and more about codestyle. I propose to keep it here to avoid unnecessary long CI checks :(